flocker/node/agents/functional/test_cinder.py

```diff
@@ -17,53 +17,42 @@
 
 from bitmath import Byte
 
-from ....testtools import skip_except
-from ..cinder import cinder_api, wait_for_volume
-from ..testtools import tidy_cinder_client_for_test
+from twisted.trial.unittest import SkipTest
+
 # make_iblockdeviceapi_tests should really be in flocker.node.agents.testtools,
 # but I want to keep the branch size down
 from ..test.test_blockdevice import (
-    make_iblockdeviceapi_tests, detach_destroy_volumes,
-    REALISTIC_BLOCKDEVICE_SIZE
+    make_iblockdeviceapi_tests,
+)
+from ..test.blockdevicefactory import (
+    ConfigMissing, ProviderType, get_blockdeviceapi_args,
+    get_blockdeviceapi_with_cleanup,
 )
+from ....testtools import REALISTIC_BLOCKDEVICE_SIZE
 
+from ..cinder import wait_for_volume
 
-def cinderblockdeviceapi_for_test(test_case, cluster_id):
+
+def cinderblockdeviceapi_for_test(test_case):
     """
     Create a ``CinderBlockDeviceAPI`` instance for use in tests.
 
     :param TestCase test_case: The test being run.
-    :param UUID cluster_id: The Flocker cluster ID for Cinder volumes.
     :returns: A ``CinderBlockDeviceAPI`` instance whose underlying
         ``cinderclient.v1.client.Client`` has a ``volumes`` attribute wrapped
         by ``TidyCinderVolumeManager`` to cleanup any lingering volumes that
         are created during the course of ``test_case``
     """
-    cinder_blockdevice_api = cinder_api(
-        cinder_client=tidy_cinder_client_for_test(test_case),
-        cluster_id=cluster_id,
-    )
-    test_case.addCleanup(detach_destroy_volumes, cinder_blockdevice_api)
-    return cinder_blockdevice_api
+    return get_blockdeviceapi_with_cleanup(test_case, ProviderType.openstack)
 
 
 # ``CinderBlockDeviceAPI`` only implements the ``create`` and ``list`` parts of
 # ``IBlockDeviceAPI``. Skip the rest of the tests for now.
-@skip_except(
-    supported_tests=[
-        'test_interface',
-        'test_created_is_listed',
-        'test_created_volume_attributes',
-        'test_list_volume_empty',
-        'test_listed_volume_attributes',
-    ]
-)
 class CinderBlockDeviceAPIInterfaceTests(
```

Comment:
It seems like `test_attach_elsewhere_attached_volume` can't obviously be run as-is against a real backend here.